File: dart/glue/lib/src/lib/bool/exists.dart
import 'package:glue/src/eval.dart';
import 'package:glue/src/ir.dart';
import 'package:glue/src/ir.dart' as ir;

final Ir exists = IrNativeFunc(existsImpl);

Eval<Ir> existsImpl(Ir arg) => Eval.pure(IrBool(ir.exists(arg)));

File: haskell/glue/src/Glue/Lib/Bool/Exists.hs
module Glue.Lib.Bool.Exists where

import Glue.Eval (Eval)
import Glue.IR as IR (IR (..), exists)

exists :: IR Eval
exists = NativeFunc existsImpl

existsImpl :: IR Eval -> Eval (IR Eval)
existsImpl = pure . Bool . IR.exists
